alacritty_terminal/src/term/mod.rs

```diff
@@ -1894,19 +1894,16 @@ impl<T: EventListener> Handler for Term<T> {
                 self.selection =
                     self.selection.take().filter(|s| !s.intersects_range(0..=last_line));
             },
-            ansi::ClearMode::Saved => {
-                let last_scrollback_line = self.grid.len() - 1;
-                if last_scrollback_line == last_line {
-                    return;
-                }
-
+            ansi::ClearMode::Saved if self.grid.len() - 1 > last_line => {
```

Comment:
Why this cryptic check rather than `self.grid.history_size() > 0`? That seems to be more clear on intent.